pkg/updater/remote_config.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote/client"
+	pbgo "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/updater/repository"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type remoteConfig struct {
+	client *client.Client
+}
+
+func newRemoteConfig(rcFetcher client.ConfigFetcher) (*remoteConfig, error) {
+	client, err := client.NewClient(
+		rcFetcher,
+		client.WithUpdater(),
+		client.WithProducts(state.ProductUpdaterCatalogDD, state.ProductUpdaterTask),
+		client.WithoutTufVerification(),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create rc client: %w", err)
+	}
+	return &remoteConfig{client: client}, nil
+}
+
+// Start starts the remote config client.
+func (rc *remoteConfig) Start(handleCatalogUpdate handleCatalogUpdate, handleRemoteAPIRequest handleRemoteAPIRequest) {
+	rc.client.Subscribe(state.ProductUpdaterCatalogDD, handleUpdaterCatalogDDUpdate(handleCatalogUpdate))
+	rc.client.Subscribe(state.ProductUpdaterTask, handleUpdaterTaskUpdate(handleRemoteAPIRequest))
+	rc.client.Start()
+}
+
+// Close closes the remote config client.
+func (rc *remoteConfig) Close() {
+	rc.client.Close()
+}
+
+// SetState sets the state of the given package.
+func (rc *remoteConfig) SetState(pkg string, state *repository.State) {
+	rc.client.SetUpdaterPackagesState([]*pbgo.PackageState{
+		{
+			Package:           pkg,
+			StableVersion:     state.Stable,
+			ExperimentVersion: state.Experiment,
+		},
+	})
+}
+
+// Package represents a downloadable package.
+type Package struct {
+	Name     string `json:"package"`
+	Version  string `json:"version"`
+	SHA256   string `json:"sha256"`
+	URL      string `json:"url"`
+	Size     int64  `json:"size"`
+	Platform string `json:"platform"`
+	Arch     string `json:"arch"`
+}
+
+type catalog struct {
+	Packages []Package `json:"packages"`
+}
+
+func (c *catalog) getPackage(pkg string, version string, arch string, platform string) (Package, bool) {
+	for _, p := range c.Packages {
+		if p.Name == pkg && p.Version == version && (p.Arch == "" || p.Arch == arch) && (p.Platform == "" || p.Platform == platform) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏Kinda wondering if we should support empty archs & platforms in the future

Reply:
Had the same thought and I concluded that we might have packages that are cross-arch compatible (ex: some tracers)